std/traits.d

```diff
@@ -5364,16 +5369,23 @@ enum bool isNumeric(T) = is(NumericTypeOf!T) && !isAggregateType!T;
 /**
 Detect whether $(D T) is a scalar type (a built-in numeric, character or boolean type).
  */
-enum bool isScalarType(T) = isNumeric!T || isSomeChar!T || isBoolean!T;
+enum bool isScalarType(T) = is(T : real);
```

Comment:
If you want to be totally safe about things, all of the traits should be tested to make sure that they don't treat `alias this` types as if they were the type that they're aliased to unless the trait is supposed to treat implicit conversions like the real thing, and very few if any of them are. Implicit conversions wreak havoc with generic code if you're not careful. And the fact that your changes to `isScalarType` didn't fail the unit tests clearly indicate that they're not thorough enough in that regard.